include/alpaka/core/Common.hpp

```diff
@@ -17,23 +17,6 @@
     #include <intrin.h>
 #endif
 
-//-----------------------------------------------------------------------------
-// Boost does not yet correctly identify clang when compiling CUDA code.
-// After explicitly including <boost/config.hpp> we can safely undefine some of the wrong settings.
-#if BOOST_COMP_CLANG_CUDA
-    #include <boost/config.hpp>
-    #undef BOOST_NO_CXX11_VARIADIC_TEMPLATES
-#endif
-
-//-----------------------------------------------------------------------------
-// Boost disables variadic templates for nvcc (in some cases because it was buggy).
-// However, we rely on it being enabled.
-// After explicitly including <boost/config.hpp> we can safely undefine the wrong setting.
-#if BOOST_COMP_NVCC
-    #include <boost/config.hpp>
```

Comment:
but we are using `BOOST_LANG_CUDA` in line 44 (27), which is defined by this include.

Reply:
update: ah, I see. we introduced `alpaka/core/BoostPredef.hpp` that contains those work-arounds and the include, right?

Reply:
Yes, we use the defines from boost.predef and not the ones from boost.config

Reply:
We do not need the variadic template workarounds for a long time now since we removed the dependency to boost.mpl long ago.

Reply:
hm, no that one does not include `boost/config.hpp`, but set the defines we need here via `boost/predef.h`.

But where do we set `BOOST_NO_CXX11_VARIADIC_TEMPLATES` now? Cannot find other occurrences in the code base.

Reply:
I added your details to the PR description, because that saves reviewers time.